src/Microsoft.AspNetCore.Routing/Matchers/JumpTableBuilder.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Routing.Matchers
+{
+    internal class JumpTableBuilder
+    {
+        private readonly List<(string text, int destination)> _entries = new List<(string text, int destination)>();
+
+        // The destination state when none of the text entries match.
+        public int DefaultDestination { get; set; } = -1;
```

Comment:
I wonder if `-1` should be placed in a const. There are some places in code below where you test if a destination is -1 and a const would help with readability compared to a magic number.

Reply:
I'm fine with it.